File: nest/src/product/product.business.ts
import { Injectable } from '@nestjs/common';
import { ProductService } from './product.service';
import { CreateProductQueryDto } from './dto/create-product-query.dto';
import { UpdateProductQueryDto } from './dto/update-product-query.dto';
import { ProductResponseDto } from './dto/product-response.dto';

@Injectable()
export class ProductBusiness {
    constructor(private readonly productService: ProductService) {}

    async create(
        createProductDto: CreateProductQueryDto,
    ): Promise<ProductResponseDto> {
        const product = await this.productService.create(createProductDto);

        return new ProductResponseDto(product);
    }

    async findAll(): Promise<ProductResponseDto[]> {
        const products = await this.productService.findAll();

        return products.map((product) => new ProductResponseDto(product));
    }

    async findOne(id: number): Promise<ProductResponseDto> {
        const product = await this.productService.findOne(id);

        return new ProductResponseDto(product);
    }

    async update(
        id: number,
        updateProductDto: UpdateProductQueryDto,
    ): Promise<ProductResponseDto> {
        const product = await this.productService.update(id, updateProductDto);

        return new ProductResponseDto(product);
    }

    async remove(id: number): Promise<void> {
        await this.productService.remove(id);
    }
}

File: spring/src/main/java/com/example/alten/product/business/ProductBusiness.java
package com.example.alten.product.business;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.alten.product.dto.CreateProductDto;
import com.example.alten.product.dto.ProductResponseDto;
import com.example.alten.product.dto.UpdateProductDto;
import com.example.alten.product.model.Product;
import com.example.alten.product.service.ProductService;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class ProductBusiness {

    private final ProductService productService;

    @Autowired
    public ProductBusiness(ProductService productService) {
        this.productService = productService;
    }

    public ProductResponseDto create(CreateProductDto createProductDto) {
        Product product = productService.create(createProductDto);

        return mapToResponseDto(product);
    }

    public List<ProductResponseDto> findAll() {
        List<Product> products = productService.findAll();

        return products.stream()
                .map(this::mapToResponseDto)
                .collect(Collectors.toList());
    }

    public ProductResponseDto findOne(Long id) {
        Product product = productService.findOne(id);

        return mapToResponseDto(product);
    }

    public ProductResponseDto update(Long id, UpdateProductDto updateProductDto) {
        Product product = productService.update(id, updateProductDto);

        return mapToResponseDto(product);
    }

    public void remove(Long id) {
        productService.remove(id);
    }

    private ProductResponseDto mapToResponseDto(Product product) {
        return new ProductResponseDto(product);
    }
}
